ontolearner/learner/taxonomy_discovery/rwthdbis.py

```diff
@@ -0,0 +1,792 @@
+
+    If no `context_json_path` is provided, the class precomputes a
+    context file ({ontology_name}_processed.json) directly from the ontology
+    object.
+    """
+
+    # Sentences containing any of these phrases are pruned from term_info.
+    _CONTEXT_REMOVALS = [
+        "couldn't find any",
+        "does not require",
+        "assist you further",
+        "feel free to",
+        "I'm currently unable",
+        "the search results",
+        "I'm unable to",
+        "recommend referring directly",
+        "bear with me",
+        "searching for the most relevant information",
+        "I'm currently checking the most relevant",
+        "already in English",
+        "require further",
+        "any additional information",
+        "already an English",
+        "don't have information",
+        "I'm sorry,",
+        "For further exploration",
+        "For more detailed information",
+    ]
+
+    def __init__(
+        self,
+        min_predictions: int = 1,
+        model_name: str = "distilroberta-base",
+        output_dir: str = "./results/{model_name}",
+        max_length: int = 256,
+        per_device_train_batch_size: int = 8,
+        gradient_accumulation_steps: int = 4,
+        num_train_epochs: int = 1,
+        learning_rate: float = 2e-5,
+        weight_decay: float = 0.01,
+        logging_steps: int = 25,
+        save_strategy: str = "epoch",
+        save_total_limit: int = 1,
+        fp16: bool = True,
+        bf16: bool = False,
+        seed: int = 42,
+        negative_ratio: int = 5,
+        bidirectional_templates: bool = True,
+        context_json_path: Optional[str] = None,
+        ontology_name: str = "Geonames"
+    ) -> None:
+        super().__init__()
+
+        self.model_name = model_name
+        self.safe_model_name = model_name.replace("/", "__")
+
+        resolved_output = output_dir.format(model_name=self.safe_model_name)
+        self.output_dir = str(Path(resolved_output))
+        Path(self.output_dir).mkdir(parents=True, exist_ok=True)
+
+        self.min_predictions = int(min_predictions)
+        self.max_length = int(max_length)
+        self.per_device_train_batch_size = int(per_device_train_batch_size)
+        self.gradient_accumulation_steps = int(gradient_accumulation_steps)
+        self.num_train_epochs = float(num_train_epochs)
+        self.learning_rate = float(learning_rate)
+        self.weight_decay = float(weight_decay)
+        self.logging_steps = int(logging_steps)
+        self.save_strategy = str(save_strategy)
+        self.save_total_limit = int(save_total_limit)
+        self.fp16 = bool(fp16)
+        self.bf16 = bool(bf16)
+        self.seed = int(seed)
+
+        self.negative_ratio = int(negative_ratio)
+        self.bidirectional_templates = bool(bidirectional_templates)
+        self.context_json_path = context_json_path
+
+        self.ontology_name = ontology_name
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
```

Comment:
What if the user decides to run on the CPU? The `device` should be something that has been specified in the argument!